### PR TITLE
Ci/actions

### PR DESCRIPTION
**Pull request by jneo8**

### CODE REVIEW

File: tests/unit/test_charm.py [hunk 1]
```diff
@@ -78,26 +78,39 @@ def test_on_config_changes_aborts_if_host_is_container(harness, mocker):
 
 
 @pytest.mark.parametrize(
-    "initiator_content, expected_initiator_name",
+    "initiator_content, expected_initiator_name, initiatorname_file",
```
Comment (Pjack) [inline tests/unit/test_charm.py:81]: What does initiator_name_file mean? It mixed the value "True" and "None" together.

**Review by Pjack — CHANGES_REQUESTED** [review]
It looks like it tried to fix two things in a PR. Could you add explanition in the PR description to explain why we need to modify the funcional test case? thanks!

File: tests/unit/test_charm.py (hunk 1 above)
Comment (jneo8) [reply tests/unit/test_charm.py:81]: It's a missing part in unit test.


There is a logic that `_get_initiator_name_from_file` function can take `None` as `initiatorname_file` input value. So Ture means using the mock value for it and `None` means pass `None` to the function.

Comment (dashmage) [reply tests/unit/test_charm.py:81]: Thanks for adding an extra test case. But I think we could use some nicer variable names and clean the logic up a bit.

```python
@pytest.mark.parametrize(
    "initiator_content, expected_initiator_name, iscsi_config_file_provided",
    [
        (None, None, True),  # file doesn't exist
        ("", None, True),  # file exists but no content
        (INITIATOR_CONTENT_TEMPLATE.format("InitiatorName=test-iqn"), "test-iqn", True),
        (INITIATOR_CONTENT_TEMPLATE.format("InitiatorName=test-iqn"), "test-iqn", False),
    ...
)
    ...
    harness.charm.ISCSI_CONF_PATH = tmp_path
    if iscsi_config_file_provided:
        initiatorname_file = tmp_path / "initiatorname.iscsi"
        if initiator_content is not None:
            initiatorname_file.write_text(initiator_content)
    else:
        harness.charm.ISCSI_INITIATOR_NAME.write_text(initiator_content)
    initiator_name = harness.charm._get_initiator_name_from_file(initiatorname_file)
    assert initiator_name == expected_initiator_name
```

* `initiatorname_file` --> `iscsi_config_file_provided` to more easily understand that this denotes whether the arg to provide the initiatorname.iscsi config file was provided or not.
* Multiple if clauses was simplified to one if/else clause.

**Review by dashmage — CHANGES_REQUESTED** [review]
Left one comment [here](https://github.com/canonical/charm-storage-connector/pull/19/files#r1464665171).

File: tests/unit/test_charm.py (hunk 1 above)
Comment (jneo8) [reply tests/unit/test_charm.py]: Updated.

**Review by dashmage — APPROVED** [review]
LGTM now, thanks :smile:

**Review by Pjack — APPROVED** [review]

**Pull request merged**